fix: catch http errors when downloading reference files

maehler opened the pull request:
### This PR:

HTTP errors caused the reference file download to halt on the first error. This PR fixes this by catching the error, logging it, adding the file in question to the list of failed files and then continue with the download.

### Review and tests: 
- [ ] Tests pass
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Code review
- [ ] `CHANGELOG.md` is updated
- [ ] New code is executed and covered by tests, and test approve

jonca79 commented:
How does the output look now when you are ready with the download and it fails? It is very clear that it failed? (Like bold red letter with a annoying sound)

maehler commented:
This is how it looks in the test for 387e069 (that I reverted) looked:

```
ERROR    failed to fetch resource at https://raw.githubuserconte/ reference.py:75
         nt.com/hydra-genetics/hydra-genetics/develop/tests/util                
         s/files/data/not_a_file.txt: HTTP Error 404: Not Found                 
ERROR    Failed to retrieve 1                                  references.py:223
ERROR    Failed: test_new/not_a_file.txt                       references.py:224
UPDATED: 9
NOT UPDATED: 1
```

In the terminal it has all the colours and stuff. It logs an error when it encounters it, and then the CLI command also prints an error when it summarises the download. In other words, even if you run it with `--debug` it should be hard to miss. It doesn't have a non-zero exit code though, but that doesn't happen for any other failure either (like mismatching checksums), so I didn't want to change that behaviour.
